public/app/features/templating/formatRegistry.ts

```diff
@@ -241,3 +248,36 @@ function encodeURIComponentStrict(str: string) {
     );
   });
 }
+
+function replaceText({ value, args }: FormatOptions) {
```

Comment:
I would like to refactor this code a bit to make it easier to reason about. I'm not sure how to do that the best way but I needed to walk through this code a couple of times before getting all the different scenarios.

Reply:
@mckn I pushed one more commit to try to improve the readability. It's quite a bit longer, but things have explicit names now and the various length permutations are more explicit cases. Let me know what you think.